pkg/services/provisioning/alerting/file/rules_types.go

```diff
@@ -81,6 +82,45 @@ type AlertRuleV1 struct {
 	IsPaused     values.BoolValue      `json:"isPaused" yaml:"isPaused"`
 }
 
+// UnmarshalJSON supports the grandfathered typo dashboardUid. This should be removed in V2.
+func (rule *AlertRuleV1) UnmarshalJSON(b []byte) error {
+	var t struct {
+		AlertRuleV1 `json:",inline" yaml:",inline"`
+		DasboardUID values.StringValue `json:"dasboardUid" yaml:"dasboardUid"`
+	}
+	if err := yaml.Unmarshal(b, &t); err != nil {
+		return err
+	}
+	*rule = t.AlertRuleV1
+
+	if rule.DashboardUID.Value() == "" && t.DasboardUID.Value() != "" {
+		rule.DashboardUID = t.DasboardUID
+	}
```

Comment:
```suggestion
	if rule.DashboardUID.Value() == "" {
		rule.DashboardUID = t.DasboardUID
	}
```
logically equivalent